driver/lib/sessions/ios.ts

```diff
@@ -35,10 +32,18 @@ export const getObservatoryWsUri = async (proxydriver) => {
   if (realDevice) {
     // @todo check if `brew install usbmuxd` is needed
     log.info(`Running on iOS real device, doing "iproxy" now`);
-    const cmd = `iproxy ${urlObject.port} ${urlObject.port} ${udid}`;
-    log.debug(cmd);
-    await execPromise(cmd);
-    log.info(`"iproxy" successfully`);
+    const args = [urlObject.port, urlObject.port, udid];
+    log.debug(`iproxy ${urlObject.port} ${urlObject.port} ${udid}`);
+    const cmd = spawn(`iproxy`, args);
+    let stdout = ``;
+    cmd.stdout.on(`data`, (data) => {
+      stdout += data;
+      if (stdout.includes(`waiting for connection`)) {
```

Comment:
It seems enough to add the following lines to detect a failure:
```
    cmd.stderr.on('data', (data) => {
      log.error(`iproxy stderr: ${data}`);
    });
```
Maybe we can just remove this `if...else...`?

Reply:
I made another attempt now: https://github.com/truongsinh/appium-flutter-driver/pull/130. Have tested on my machine and it works fine.